### PR TITLE
feat: separate `KromaOutputV0` from `OutputV0`

### PR DESCRIPTION
**Pull request by seolaoh**
# Description

Separated `KromaOutputV0` from `OutputV0` so that the former is used before `mpt_time`, the latter is used after `mpt_time`. Also separated `OutputWithProofResponse` from `OutputResponse`, since `OutputWithProofResponse` will not be used anymore after the transition to zkVM.

There are some codes to be uncommented after geth version updated to the state migration version. I tested `proveWithdrawal` in MPT by manually setting `kromaChainConfig.Zktrie = false`.

### CODE REVIEW

File: op-e2e/actions/user_test.go [hunk 1]
```diff
@@ -125,6 +133,9 @@ func runCrossLayerUserTest(gt *testing.T, test hardforkScheduledTest) {
 	if test.ecotoneTime != nil {
 		require.Zero(t, uint64(*test.ecotoneTime)%uint64(dp.DeployConfig.L2BlockTime), "ecotone fork must be aligned")
 	}
+	if test.kromaMPTTime != nil {
+		require.Zero(t, uint64(*test.kromaMPTTime)%uint64(dp.DeployConfig.L2BlockTime), "kroma mpt fork must be aligned")
+	}
```
Comment (seolaoh) [inline op-e2e/actions/user_test.go:138]: @Pangssu I just added this, but I wonder if kromaMPTTime should be set to exactly the timestamp of specific L2 block or can be the middle timestamp between two L2 blocks.

**Comment by kangsorang** [comment]
One suggestion that just occurred to me: what if the output is differentiated by defining a version field rather than using the presence/absence of a field value?

File: op-e2e/actions/user_test.go (hunk 1 above)
Comment (Pangssu) [reply op-e2e/actions/user_test.go:138]: It must be the exact timestamp of the L2 block.

File: kroma-chain-ops/genesis/genesis.go [hunk 2]
```diff
@@ -62,7 +62,9 @@ func NewL2Genesis(config *DeployConfig, block *types.Block) (*core.Genesis, erro
 		ShanghaiTime:                  config.CanyonTime(block.Time()),
 		CancunTime:                    config.EcotoneTime(block.Time()),
 		EcotoneTime:                   config.EcotoneTime(block.Time()),
-		InteropTime:                   config.InteropTime(block.Time()),
+		// TODO(seolaoh): uncomment this when geth updated
+		// KromaMPTTime:                  config.KromaMPTTime(block.Time()),
```
Comment (Pangssu) [inline kroma-chain-ops/genesis/genesis.go:66]: I named it `KromaMptTime`

File: op-node/rollup/types.go [hunk 3]
```diff
@@ -88,6 +88,10 @@ type Config struct {
 	// Active if EcotoneTime != nil && L2 block timestamp >= *EcotoneTime, inactive otherwise.
 	EcotoneTime *uint64 `json:"ecotone_time,omitempty"`
 
+	// KromaMPTTime sets the activation time of the Kroma MPT network upgrade.
+	// Active if KromaMPTTime != nil && L2 block timestamp >= *KromaMPTTime, inactive otherwise.
+	KromaMPTTime *uint64 `json:"kroma_mpt_time,omitempty"`
```
Comment (Pangssu) [inline op-node/rollup/types.go:93]:
```suggestion
	KromaMptTime *uint64 `json:"mpt_time,omitempty"`
```

**Comment by seolaoh** [comment]
> One suggestion that just occurred to me: what if the output is differentiated by defining a version field rather than using the presence/absence of a field value?

The version field still exists, but just omitting it means that the version is zero which we only have currently.

File: op-e2e/actions/user_test.go (hunk 1 above)
Comment (seolaoh) [reply op-e2e/actions/user_test.go:138]: It's just for my curiosity, why should it be the exact timestamp?

**Review by Pangssu — APPROVED** [review]

File: kroma-chain-ops/genesis/genesis.go (hunk 2 above)
Comment (seolaoh) [reply kroma-chain-ops/genesis/genesis.go:66]: Why not using `KromaMPTTime` following the [go naming convention](https://go.dev/wiki/CodeReviewComments#initialisms) for initialisms?

**Review by 0xbenyun — APPROVED** [review]

File: op-node/rollup/types.go (hunk 3 above)
Comment (seolaoh) [reply op-node/rollup/types.go:93]: I thought `kroma_mpt_time` is better since it's easy to recognize that it's kroma's own config. wdyt?

**Review by kangsorang — APPROVED** [review]

**Pull request merged**